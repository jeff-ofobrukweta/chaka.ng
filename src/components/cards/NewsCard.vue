<template>
    <div class="card-news">
        <img
            class="card-news__img"
            src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
            :alt="news.title"
        />
        <div class="card-news__title">
            <h5>{{ news.title | truncate(48) }}</h5>
            <!-- <h5>{{ line1 }}</h5>
      <h5 class="card-news__ellipsis">{{ line2 }}</h5> -->
        </div>
    </div>
</template>

<script>
export default {
    name: 'news-card',
    props: {
        news: {
            type: Object,
            required: true
        }
    },
    computed: {
        line1() {
            return this.news.title.substring(0, 20);
        },
        line2() {
            return this.news.title.substring(20);
        }
    }
};
</script>
