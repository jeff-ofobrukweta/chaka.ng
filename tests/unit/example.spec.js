import { shallowMount,mount } from '@vue/test-utils';
import NewsCard from '@/components/cards/NewsCard.vue';



/*
Inputs

Data: the component’s data
Props: the component’s props
User Interaction: user clicks a button
Lifecycle Methods: mounted(), created(), etc.
Store: state management data
Route Params: data in the route
*/



// test the news card and check for images link is valid
describe('NewsCard-vue', () => {
    it('snapshot renders props.news when passed', () => {
        const news = {
            imageUrl:"https://chaka.ng/dashboard/singlestock/BABA",
            name:"news",
            title:"title"
        }
        const wrapper = mount(NewsCard,{
            propsData: { news}
          })
          expect(wrapper).toMatchSnapshot()
    });
});

describe('Validate the news object.vue', () => {
    it('renders props.msg when passed', () => {
        const news = {
            imageUrl:"https://chaka.ng/dashboard/singlestock/BABA",
            name:"news",
            title:"title"
        }
      const wrapper = shallowMount(NewsCard, {
        propsData: { news }
      })
      console.log('boom',wrapper.text())
      expect(wrapper.text()).toMatch(news)
    })
  })


