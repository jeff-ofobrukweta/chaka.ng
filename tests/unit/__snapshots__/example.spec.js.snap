// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard-vue snapshot renders props.news when passed 1`] = `
<a target="_blank" rel="noreferrer noopener" class="card-news"><img src="https://chaka.ng/dashboard/singlestock/BABA" alt="title" class="card-news__img">
  <div class="card-news__title">
    <h5>news</h5>
  </div>
</a>
`;
